main/lsp/requests/rename.cc

```diff
@@ -148,14 +168,24 @@ class MethodRenamer : public Renamer {
 
 private:
     string replaceMethodNameInDef(string def) {
-        // We can't do a simple string replace-all because of cases like:
+        const vector<string> unsupportedDefPrefixes{"attr_reader", "attr_accessor", "attr_writer"};
+        for (auto u : unsupportedDefPrefixes) {
+            if (absl::StartsWith(def, u)) {
+                invalid = true; // ensures the entire rename operation is blocked, not just this particular location
+                error = "Attribute method renames are unsupported.";
```

Comment:
WDYT about a message like:

```suggestion
                error = fmt::format("Sorbet does not support renaming `{}`s", u);
```